### ISSUE TITLE
How to pass through OAUTH 2.0

### ISSUE THREAD

**Issue by DomV-cloud** [jstedfast/MailKit#1418]
Hi,
I am developing an application(In WPF) that reads emails from Outlook mailbox, then extracts the data, where I work with them further. I have two emails, one is a test one, where AOUTH 2.0 is not needed for verification. From that email, I get the emails the way I want. But then I have a company one, where AOUTH 2.0 is enabled and I can't get emails anymore. I have implemented your code to fix the authentication, but the problem is that nothing is displayed. So I don't know where to look for what is wrong. The application does not log in (I checked it in Azure), but I followed the official Microsoft documentation when I registered. Of course, I also tried ProtocolLogger. Nothing is displayed at all. My example to show my problem is in Console Application. Thanks for any advice.

MailKit version 3.3.0
C# WPF 
Windows 11

**Expected behavior**
Read emails from mailbox with enabled AOUTH 2.0

```
  public async void OAUTH(EmailParser emailParser)
        {
            var options = new PublicClientApplicationOptions
            {
                ClientId = "ID",
                TenantId = "ID",
                RedirectUri = "https://login.microsoftonline.com/common/oauth2/nativeclient"
            };

            var publicClientApplication = PublicClientApplicationBuilder
                .CreateWithApplicationOptions(options)
                .Build();
         

            var scopes = new string[] {
    "mail",
    "offline_access",
    "https://outlook.office.com/IMAP.AccessAsUser.All", // Only needed for IMAP
    //"https://outlook.office.com/POP.AccessAsUser.All",  // Only needed for POP
    //"https://outlook.office.com/SMTP.Send", // Only needed for SMTP
};

            var authToken = await publicClientApplication.AcquireTokenInteractive(scopes).ExecuteAsync();

            var oauth2 = new SaslMechanismOAuth2(authToken.Account.Username, authToken.AccessToken);

            using (var client = new ImapClient(new ProtocolLogger("imap.log")))
            {
                using (var cancel = new CancellationTokenSource())
                {

                await client.ConnectAsync("outlook.office365.com", 993, SecureSocketOptions.SslOnConnect);
                await client.AuthenticateAsync(oauth2);

                 //client.Authenticate(emailParser.Username, emailParser.Password, cancel.Token);


                    var inbox = client.Inbox;
                    inbox.Open(FolderAccess.ReadOnly, cancel.Token);

                    Console.WriteLine("Total messages: {0}", inbox.Count);
                    Console.WriteLine("Recent messages: {0}", inbox.Unread);

                    for (int i = 0; i < inbox.Count; i++)
                    {
                        var message = inbox.GetMessage(i, cancel.Token);
                        Console.WriteLine(message.TextBody);
                    }
                }
            }
        }
```

**Comment by ekalchev** [comment]
It looks ok, probably you missed to configure something in Microsoft Graph. It is not problem with MailKit for sure because we use it with OAuth and Outlook

**Comment by DomV-cloud** [comment]
> It looks ok, probably you missed to configure something in Microsoft Graph. It is not problem with MailKit for sure because we use it with OAuth and Outlook

Hi [ekalchev](https://github.com/ekalchev),
Thanks for your comment.  So I'll try to register the application again or check the instructions. I have one more question. Why is nothing displayed? Not even ProtocolLogger?

**Comment by jstedfast** [comment]
If nothing is logged, then it is probably failing to connect to outlook.office365.com (the first log message happens only after connecting)
